test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,397 @@
+				Elasticsearch: v1.ElasticsearchSpec{
+					NodeCount: 1,
+					Resources: &corev1.ResourceRequirements{
+						Limits:   corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
+						Requests: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
+					},
+				},
+			},
+			Query: v1.JaegerQuerySpec{
+				Options: v1.NewOptions(map[string]interface{}{
+					"es.version":                     "5",
+					"query.bearer-token-propagation": "true",
+					"es.tls":                         "false",
+				}),
+			},
+			Ingress: v1.JaegerIngressSpec{
+				Options: v1.NewOptions(map[string]interface{}{
+					"pass-access-token":      "true",
+					"pass-user-bearer-token": "true",
+					"scope":                  "user:info user:check-access user:list-projects",
+					"pass-basic-auth":        "false",
+				}),
+			},
+		},
+	}
+	return exampleJaeger
+}
+
+func (suite *TokenPropagationTestSuite) deployJaegerWithPropagationEnabled() {
+	queryName := fmt.Sprintf("%s-query", name)
+	collectorName := fmt.Sprintf("%s-collector", name)
+
+	// create jaeger custom resource
+	suite.exampleJaeger = getESJaegerInstance()
+	err := fw.Client.Create(goctx.TODO(),
+		suite.exampleJaeger,
+		&framework.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       timeout,
+			RetryInterval: retryInterval,
+		})
+	require.NoError(t, err, "Error deploying example Jaeger")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, collectorName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for collector deployment")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, queryName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for query deployment")
+
+	route := findRoute(t, fw, name)
+
+	suite.host = route.Spec.Host
+	suite.queryServiceEndPoint = fmt.Sprintf("https://%s/api/traces?service=%s", suite.host, testServiceName)
+
+}
+
+func TestTokenPropagationSuite(t *testing.T) {
+	suite.Run(t, new(TokenPropagationTestSuite))
+}
+
+func newHTTPSClient() *http.Client {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{
+		Transport: tr,
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+	return client
+}
+
+func oAuthAuthorization(host, user, pass string) (*http.Client, error) {
+	/* Setup client*/
+	cookieJar, _ := cookiejar.New(nil)
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+		Jar: cookieJar,
+	}
+	/* Start oauth */
+	resp, err := client.Get("https://" + host + "/oauth/start")
+	defer resp.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+	responseBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	var req *http.Request
+	/* Submit form */
+	if hasForm(responseBytes) {
+		req = getLoginFormRequest(responseBytes, resp.Request.URL, user, pass)
```

Comment:
For OpenShift, do you really need to parse the form and simulate a real user there? Can't you just send the user:pass with basic auth? I think the Maistra components to send the requests via basic auth.

Reply:
Well, I can get a token an pass it to query service, but I don't know how to do it and test the whole thing using oauth proxy sidecar, Can we use basic authentication to and set the oauth proxy cookie? May be I'm missing something. Or we don't need to test using the sidecar?

Reply:
Maistra QE can confirm, but I understand that you can send basic auth and the proxy will do all the token negotiation based on it. Jaeger will see a token, just like if it was a browser session.